pkgs/development/libraries/mesa/default.nix

```diff
@@ -144,7 +145,8 @@ self = stdenv.mkDerivation {
     ++ lib.optionals stdenv.isLinux [ libomxil-bellagio libva-minimal ]
     ++ lib.optionals stdenv.isDarwin [ libunwind ]
     ++ lib.optionals enableOpenCL [ libclc llvmPackages.clang llvmPackages.clang-unwrapped ]
-    ++ lib.optional withValgrind valgrind-light;
+    ++ lib.optional withValgrind valgrind-light
+    ++ lib.optional (elem "zink" galliumDrivers) vulkan-loader;
```

Comment:
I was confused by this, because according to Mesa's meson option definitions, `-Dgallium-drivers=auto` is supposed to build all drivers compatible with the architecture.  Having looked into it, `auto` never includes `zink`, so we don't have to worry about adding it in the `auto` case on the right platform.

I think there should be a comment explaining this so that future readers don't think they've spotted a bug like I did.

Reply:
I added a comment. gallium-drivers=auto is really annoying (e.g. it won't build r600 for AMD GPUs on ARM even though there are ARM SoCs with RDNA2 GPUs and you can put an AMD GPU on an ARM desktop just fine)